opendatahub-io/odh-dashboard
# Refactor BYON images table

Closes #1432 

## Description


1. Refactor the whole table, using the generic table component
2. use `useFetchState` hook to get BYON images
3. Refactor the usage of the modals. (as for the refactor of the contents in the modals, it will be solved in #1433)

<img width="1788" alt="Screenshot 2023-07-12 at 10 47 57 AM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/eb92f115-b72d-4b0d-92d6-3aa51964b597">

## How Has This Been Tested?



1. Go to the settings -> Notebook image settings
2. Create a notebook image, and try to play around with it, like editing, deleting, toggle the enablement... make sure nothing is broken

## Test Impact


TBD

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
